docs/instant.md

```diff
@@ -7,17 +7,80 @@
 
 A `Temporal.Instant` is a single point in time (called **"exact time"**), with a precision in nanoseconds.
 No time zone or calendar information is present.
-As such `Temporal.Instant` has no concept of days, months or even hours.
+To obtain local date/time units like year, month, day, or hour, a `Temporal.Instant` must be combined with a `Temporal.TimeZone` instance or a time zone string.
 
-For convenience of interoperability, it internally uses nanoseconds since the [Unix epoch](https://en.wikipedia.org/wiki/Unix_time) (midnight UTC on January 1, 1970).
-However, a `Temporal.Instant` can be created from any of several expressions that refer to an exact time, including an ISO 8601 string with a time zone such as `'2020-01-23T17:04:36.491865121-08:00'`.
+<!-- prettier-ignore-start -->
+```javascript
+instant = Temporal.Instant.from('2020-01-01T00:00+05:30'); // => 2019-12-31T18:30:00Z
+instant.epochNanoseconds; // => 1577817000000000000n
+
+// `Temporal.Instant` lacks properties that depend on time zone or calendar
+instant.year; // => undefined
 
-If you have a legacy `Date` instance, you can use its `toTemporalInstant()` method to convert to a `Temporal.Instant`.
+zdtTokyo = instant.toZonedDateTimeISO('Asia/Tokyo'); // => 2020-01-01T03:30:00+09:00[Asia/Tokyo]
+zdtTokyo.year; // => 2020
+zdtTokyo.toPlainDate(); // => 2020-01-01
+
+tzLA = Temporal.TimeZone.from('America/Los_Angeles');
+zdtLA = instant.toZonedDateTimeISO(tzLA); // => 2019-12-31T10:30:00-08:00[America/Los_Angeles]
+zdtLA.year; // => 2019
+zdtLA.toPlainDate(); // => 2019-12-31
+```
+<!-- prettier-ignore-end -->
 
-Since `Temporal.Instant` doesn't contain any information about time zones, a `Temporal.TimeZone` is needed in order to convert it into a `Temporal.ZonedDateTime` or `Temporal.PlainDateTime` (and from there into any of the other `Temporal` objects.)
+`Temporal.Instant` stores a count of integer nanoseconds since the [Unix epoch](https://en.wikipedia.org/wiki/Unix_time): midnight UTC on January 1, 1970.
+For interoperability with `Date` and other APIs, `Temporal.Instant` also offers conversion properties and methods for seconds, milliseconds, or microseconds since epoch.
+A `Temporal.Instant` can also be created from an ISO 8601 / RFC 3339 string like `'2020-01-23T17:04:36.491865121-08:00'` or `'2020-01-24T01:04Z'`.
 
 Like Unix time, `Temporal.Instant` ignores leap seconds.
 
+## Interoperability with `Date`
+
+`Temporal.Instant` is the easiest way to interoperate between `Temporal` objects and code using `Date`.
+Because `Date` and `Temporal.Instant` both use a time-since-epoch data model, conversions between them are zero-parameter method calls that are lossless (except sub-millisecond precision is truncated when converting to `Date`).
+
+<!-- prettier-ignore-start -->
```

Comment:
Maybe we could remove these examples and have a link to the cookbook (https://tc39.es/proposal-temporal/docs/cookbook.html#converting-between-temporal-types-and-legacy-date) here?

Reply:
Yeah, I was wondering about that too.  That `Date`-interop cookbook content (which is really well-written... nice work on that!) is accurate and helpful but too long to put here. And I think we know enough about human behavior to know that if we just link to the cookbook content, most users won't read it. 

So I was trying to figure out what information should go here in the main docs to explain the most important concepts related to `Date` interop, and to prevent the common bugs that stem from misunderstanding these concepts. I think these concepts are: 
1. Even though `Date` seems like it stores a time zone & offset, it's really just msecs since epoch. The timezone and offset in `toString`, `getFullYear`, etc. are just runtime illusions!
2. `Date`, confusingly, has both local-timezone and UTC functionality, and mixing them up can ruin your day.  This is especially a problem when storing data that's not an exact time, like date-only values.
3. `Instant` and `Date` represent the same data at different precision. Therefore, `Temporal` and `Date` interop usually involves `Instant` as a bridge.

I wrote several different versions of content to try to explain those three concepts in words only, but ultimately I found it more concise to explain them using a mix of words and code.  Each code sample replaced 1-2 paragraphs of text.

So in my mind, the samples here are less about showing how to do a specific task (which would belong in the cookbook) but rather as a shorter and clearer alternative to more text. 

So I'd prefer to leave these samples in. What do you think?

Regardless, you're right that we should also be linking to the cookbook here. So I just pushed an update to add a link.

Reply:
Well, my concern is that this seems to cover a lot of the same material as the cookbook recipe. I think you're right that people would be less likely to click through to the cookbook, but on the other hand, if we have similar content here then they're more likely to just copy and paste this, and _definitely_ not click through to the cookbook. So I feel that if we include full examples here at the top of the page, they'll be cookbook material whether we like it or not.

How about something like this? I wouldn't mind what you have, but this to me covers your three points without becoming a copy and paste temptation.

> `Temporal.Instant` is the easiest way to interoperate between `Temporal` objects and code using `Date`. Because `Date` and `Temporal.Instant` both use a time-since-epoch data model, conversions between them are zero-parameter method calls that are lossless (except sub-millisecond precision is truncated when converting to `Date`).
>
> ```javascript
> // Convert from `Temporal.Instant` to `Date` (which uses millisecond precision)
> instant = Temporal.Instant.from('2020-01-01T00:00:00.123456789+05:30');
>                                                            // => 2019-12-31T18:30:00.123456789Z
> date = new Date(instant.epochMilliseconds);
> date.toISOString();                                        // => 2019-12-31T18:30:00.123Z
> ```
>
> `Date`, like `Temporal.Instant`, only stores an integer count of time since epoch. It does not store a time zone nor an offset. The time zone and offset shown below are fetched (from the caller's time zone) at runtime by `Date.prototype.toString`.
> ```javascript
> date.toString(); // => Tue Dec 31 2019 10:30:00 GMT-0800 (Pacific Standard Time)
> ```
> Be aware that this is a common source of bugs when using `Date`! For more information on how to safely convert from `Temporal.Instant` and other Temporal types to legacy `Date`, see [Legacy Date from Temporal types](https://tc39.es/proposal-temporal/docs/cookbook.html#legacy-date-from-temporal-types) in the cookbook.
>
> After a `Date` has been converted to a `Temporal.Instant`, it's easy to convert to other `Temporal` types. Most of these conversions will go via `Temporal.ZonedDateTime`, so make sure to use the correct time zone when converting between `Temporal.Instant` and other `Temporal` types. See [Temporal types from legacy Date](https://tc39.es/proposal-temporal/docs/cookbook.html#temporal-types-from-legacy-date) in the cookbook for examples.
>
> Especially using a `Date` to represent a year/month/day without a time, is another common source of bugs with `Date`. See [our new cookbook example]() for how to deal with this safely when interoperating between Temporal and `Date`.
>
> For detailed how-to information about interoperating with `Date` in general, see [Converting between `Temporal` types and legacy `Date`](cookbook.md#converting-between-temporal-types-and-legacy-date).

Reply:
I like where you're going here: reduce code without getting too wordy, and add more pointers to the cookbook. I'll riff on this and we can review again.